src/compiler/error-detector.js

```diff
@@ -96,3 +94,27 @@ function checkExpression (exp: string, text: string, errors: Array<string>) {
     }
   }
 }
+
+function isValidExpression (exp: string) {
+  try {
+    new Function(`return ${exp}`)
+    return true
+  } catch (e) {
+    /* istanbul ignore next */
+    try {
+      const babylon = (new Function(`return require('babylon')`))()
+      /* $flow-disable-line */
+      babylon.parseExpression(exp, {
+        plugins: [
+          'objectRestSpread',
+          'functionBind',
+          'numericSeparator',
+          'optionalChaining'
+        ]
```

Comment:
@chrisvfritz, plugins are a part of `babylon` and are bundled with it.

Looking for `.babelrc` synchronously in the directory tree would be very expensive and blocking. Specially if we consider that sometimes you need to go a lot of directories up to find the `.babelrc`, and we cannot do that asynchronously :(